pkg/controller/replicaset/replica_set.go

```diff
@@ -181,6 +181,27 @@ func (rsc *ReplicaSetController) getPodReplicaSets(pod *v1.Pod) []*extensions.Re
 	return rss
 }
 
+// resolveControllerRef returns the controller referenced by a ControllerRef,
+// or nil if the ControllerRef could not be resolved to a matching controller
+// of the corrrect Kind.
+func (rsc *ReplicaSetController) resolveControllerRef(namespace string, controllerRef *metav1.OwnerReference) *extensions.ReplicaSet {
+	// We can't look up by UID, so look up by Name and then verify UID.
+	// Don't even try to look up by Name if it's the wrong Kind.
+	if controllerRef.Kind != controllerKind.Kind {
```

Comment:
> @liggitt i thought about running an offline upgrade script to modify the objects in etcd directly to update the versions in ownerReferences. How's that sound?

That sounds like it won't work well if we surface objects via two API versions at once (which we will). It adds one more touchpoint that everything that interacts with an object has to agree on an API version (or be ambiguous, which is bad in a different way)